### PR TITLE
feat: customersTable

### PR DESCRIPTION
**Pull request by kohiruim**

### CODE REVIEW

File: src/features/clientsTable/ui/ClientsTable.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  firebaseKeys,
+} from "@/shared";
+import { headerItems } from "@/shared/config";
+import { headerHandleClick } from "../model";
+
+export const ClientsTable = () => {
+  const [rowsClientsTable, setRowsClientsTable] = useState<Array<UserData>>([]);
+  const buttonSortParams = useSortedData(setRowsClientsTable);
+
+  const rows = rowsClientsTable.map((item: UserData) => (
```
Comment (Emptyyw) [inline src/features/clientsTable/ui/ClientsTable.tsx]: Попробуй это перенести в разметку

File: src/features/clientsTable/ui/ClientsTable.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    <ClientsTableRow element={item} key={item.id} />
+  ));
+
+  const header = headerItems.map(item => (
```
Comment (Emptyyw) [inline src/features/clientsTable/ui/ClientsTable.tsx]: И это

File: src/app/routes/router.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import { RequerAuth } from "./RequerAdmin/RequerAdmin";
+import "@mantine/notifications/styles.css";
+
+export const router = createBrowserRouter([
+  {
+    path: "/",
+    element: <LoginPage />,
+  },
+  {
+    path: "/users",
```
Comment (remadyt) [inline src/app/routes/router.tsx]: use enum for paths

File: src/features/addClient/model/model.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+interface Values {
+  fullname: string;
+  username: string;
+  phone: string;
+  email: string;
+  role: UserRole;
+}
+
+const AddClient = async (
```
Comment (remadyt) [inline src/features/addClient/model/model.ts]: why big letter?

Comment (remadyt) [reply src/features/addClient/model/model.ts]: maybe use createAsyncThunk

File: src/features/addClient/ui/AddClientButton.tsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    mode: "uncontrolled",
+    initialValues: {
+      fullname: "",
+      username: "",
+      phone: "",
+      email: "",
+      role: UserRole.User,
+    },
+    validate: {
+      email: (value: string) =>
```
Comment (remadyt) [inline src/features/addClient/ui/AddClientButton.tsx]: use yup or zod for validations

File: src/features/addClient/ui/AddClientButton.tsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        !emailRegex.test(value) ? "только через google почту" : null,
```
Comment (remadyt) [inline src/features/addClient/ui/AddClientButton.tsx]: add translates?

File: src/features/addClient/ui/AddClientButton.tsx [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+          <Flex justify="flex-end">
+            <Button type="submit" color="indigo">
+              Добавить клиента
+            </Button>
+          </Flex>
+        </form>
+      </Modal>
+
+      <Button color="indigo" onClick={open}>
+        Добавить клиента
```
Comment (remadyt) [inline src/features/addClient/ui/AddClientButton.tsx]: add translates here and all such places

File: src/features/clientsTable/ui/ClientsTable.tsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+  ButtonSort,
+  ClientsTableRow,
+  useSortedData,
+  firebaseKeys,
+  UserContext,
+  headerItems,
+} from "@/shared";
+import { headerHandleClick } from "../model";
+
+export const ClientsTable = () => {
```
Comment (remadyt) [inline src/features/clientsTable/ui/ClientsTable.tsx:14]: create reusable component

File: src/pages/clientsPage/model/model.ts [hunk 9]
```diff
@@ -0,0 +1,49 @@
+import type { Dispatch, SetStateAction } from "react";
+import type { UserData, SearchButtonType } from "@/shared";
+
+const search = (
```
Comment (remadyt) [inline src/pages/clientsPage/model/model.ts]: use more informative naming

File: src/pages/clientsPage/model/model.ts [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+};
+
+export const handleSubmitForm = (
+  values: FormParams,
+  setTypeSearchButton: Dispatch<SetStateAction<SearchButtonType>>,
+  tableData: TableData
+) => {
+  const { users, setUsers } = tableData;
+
+  if (values.username || values.fullname || values.phone) {
```
Comment (remadyt) [inline src/pages/clientsPage/model/model.ts]: need implement search with firebase, not only in frontend

File: src/pages/clientsPage/ui/ClientsPage.tsx [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+              label="ФИО"
+              key={form.key("fullname")}
+              {...form.getInputProps("fullname")}
+            />
+            <TextInput
+              label="Никнейм"
+              key={form.key("username")}
+              {...form.getInputProps("username")}
+            />
+            <InputBase
```
Comment (remadyt) [inline src/pages/clientsPage/ui/ClientsPage.tsx]: create reusable maskedInput

Comment (remadyt) [reply src/pages/clientsPage/ui/ClientsPage.tsx]: why need login by phone number? in spec - only with google

File: src/shared/config/clientsTable.ts [hunk 12]
```diff
@@ -0,0 +1,21 @@
+import type { KeysUserData } from "@/shared";
+
+export const headerItems = [
```
Comment (remadyt) [inline src/shared/config/clientsTable.ts]: translate?

**Review by remadyt — CHANGES_REQUESTED** [review]

File: src/features/addClient/model/model.ts [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+    }
+  }
+};
+
+export const handleAddClient = (
+  values: Values,
+  setUsers: Dispatch<SetStateAction<Nullable<Array<UserData>>>>,
+  closeWindow: () => void
+) => {
+  void AddClient(values, setUsers, closeWindow);
```
Comment (suveiko) [inline src/features/addClient/model/model.ts]: look strange, why you use `void`?

**Pull request merged**